app/connectors/ValidationConnector.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.FrontendAppConfig
+import models.upscan.UpscanTimeoutException
+import models.{UploadSubmissionValidationFailure, UploadSubmissionValidationResult, UploadSubmissionValidationSuccess, ValidationErrors}
+import org.slf4j.LoggerFactory
+import play.api.http.Status.OK
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ValidationConnector @Inject() (http: HttpClient, config: FrontendAppConfig) {
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  val url = s"${config.mdrUrl}/mandatory-disclosure-rules/validate-upload-submission"
+
+  def sendForValidation(upScanUrl: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Either[ValidationErrors, Boolean]]] =
+    http
+      .POSTString[HttpResponse](url, upScanUrl)
+      .map {
+        response =>
+          response.status match {
+            case OK =>
+              response.json.as[UploadSubmissionValidationResult] match {
+                case x: UploadSubmissionValidationSuccess =>
+                  Some(Right(x.boolean))
+                case x: UploadSubmissionValidationFailure =>
+                  Some(Left(x.validationErrors))
+              }
+          }
+      }
+      .recover {
+        case e: Throwable if e.getMessage contains "Invalid XML" =>
+          logger.warn(s"XML parsing failed. The XML parser in mandatory-disclosure-rules backend has thrown the exception: $e")
+          None
+        case e: Throwable =>
+          logger.warn(s"Remote service timed out. The XML parser in mandatory-disclosure-rules backend backend has thrown the exception: $e")
+          throw new UpscanTimeoutException
+      }
```

Comment:
I think you need to have sealed trait Error  
ValidationErrors extend the Error
Ther recover block you can define 2 new Errors types specific to the context  
Just have Either[Error, Boolean]

Reply:
Yeah I'm not sure either

Reply:
sealed trait Errors 
case class ValidationErrors(errors: Seq[String], boolean: Option[Boolean]) extend Errors
case obect InVavlidXmlError extend Erros
case class NonFatalErors(exception:Exception) extend Errors 

object ValidationErrors {
  implicit val format = Json.format[ValidationErrors]
}

-----------------
 .recover {
        case NonFatal(e) =>  if (e.getMessage contains "Invalid XML" ) {
                  logger.warn(s"XML parsing failed. The XML parser in mandatory-disclosure-rules backend has thrown the exception: $e")
        Left(InVavlidXmlError)  
        }
        else {
    
          logger.warn(s"Remote service timed out. The XML parser in mandatory-disclosure-rules backend backend has thrown the exception: $e")
          Left(NonFatalErors(e))  
        
      }
-----------------

Reply:
> 
sealed trait Errors 
case class ValidationErrors(errors: Seq[String], boolean: Option[Boolean]) extend Errors
case object InVavlidXmlError extends Errors
case class NonFatalErors(e:Exception) extends Errors 
object ValidationErrors {
  implicit val format = Json.format[ValidationErrors]
}

-----------------------------------------
 .recover {
        case NonFatal(e) =>  if (e.getMessage contains "Invalid XML" ) {
                  logger.warn(s"XML parsing failed. The XML parser in mandatory-disclosure-rules backend has thrown the exception: $e")
        Left(InVavlidXmlError)  
        }else {
          logger.warn(s"Remote service timed out. The XML parser in mandatory-disclosure-rules backend backend has thrown the exception: $e")
          Left(NonFatalErors(e))  
      }

Reply:
Also catching Throwable is not right for the same

Reply:
I've updated this now.